EarlGreySnapshots/Classes/SnapshotAssertion.swift

```diff
@@ -1,29 +1,34 @@
-//
-//  Created by Jakub Turek on 19.05.2017.
-//  Copyright © 2017 EL Passion. All rights reserved.
-//
-
 import EarlGrey
 
 func grey_snapshot(testCaseName: String?, recordMode: Bool, deviceAgnostic: Bool) -> GREYAssertionBlock {
     let testCase = testCaseName ?? CurrentTestObserver.shared.currentCase?.name ?? "-[UnknownTestCase unknownMethod]"
     let testInfo = TestNameParser().parse(testName: testCase)
+    let bundleNameProvider = BundleNameProvider(bundle: Bundle.testBundle)
+    let testName = [bundleNameProvider.name, testInfo.testClassName].flatMap { $0 }.joined(separator: ".")
 
-    return grey_snapshot(testName: testInfo.testClassName,
+    return grey_snapshot(testName: testName,
                          snapshotName: testInfo.testMethodName,
                          recordMode: recordMode,
                          deviceAgnostic: deviceAgnostic)
 }
 
-func grey_snapshot(testName: String, snapshotName: String, recordMode: Bool, deviceAgnostic: Bool,
-                   controllerFactory: SnapshotControllerCreating? = nil) -> GREYAssertionBlock {
+func grey_snapshot(testName: String,
+                   snapshotName: String,
+                   recordMode: Bool,
+                   deviceAgnostic: Bool,
+                   controllerFactory: SnapshotControllerCreating? = nil,
+                   imageDirectoryProvider: ImagesDirectoryProviding? = nil) -> GREYAssertionBlock {
     return GREYAssertionBlock.init(name: "snapshot", assertionBlockWithError: { element, errorOrNil -> Bool in
         guard let view = element as? UIView else { return false }
 
-        let factory = controllerFactory ?? SnapshotControllerFactory(environment: ProcessInfo.processInfo.environment)
-        let testController = factory.makeSnapshotController(testName: testName,
-                                                            deviceAgnostic: deviceAgnostic,
-                                                            recordMode: recordMode)
+        let factory = controllerFactory ?? SnapshotControllerFactory()
+        let directoryProvider = imageDirectoryProvider ?? ImagesDirectoryProvider()
```

Comment:
For both `factory` and `directoryProvider` you can specify defaults in the method definition 😉